accepted/system-range/system-range.md

```diff
@@ -0,0 +1,312 @@
+# System.Range
+
+**PM** [Immo landwerth](https://github.com/terrajobst)
+
+The C# team is considering providing language syntax for specifying ranges to
+represent parts of a string, array, or span. In order to make this work, the
+language has to be bound to a .NET type, which we call `System.Range`.
+
+## Scenarios and User Experience
+
+### Construcing ranges
+
+```csharp
+// Ranges are constructed with an index and a length:
+
+var fiveToTen = new Range(index: 5, length: 6);
+
+// By using a factory method, they can also be constructed from two indices:
+
+var alsoFiveToTen = Range.Between(5, 11);
+
+// The upper bound is considered exclusive so this will print 1, 2, 3:
+
+foreach (var value in Range.Between(1, 4))
+    Console.WriteLine(value);
+```
+
+The C# syntax for ranges isn't finalized and is actively worked on. While this
+is part of a separate design group it makes sense to introduce the syntax here
+so that we can look at source code as it's likely written:
+
+```csharp
+var fiveToTen = 5..11; // Equivalent to Range.Between(5, 11)
+
+// You can also construct ranges with index and length:
+var fiveToTen = 5.:6; // Equivalent to new Range.Between(5, 6)
```

Comment:
‘new Range.Between’?

Reply:
No, `Between` is a static factory method, hence no `new`.

Reply:
Ah, never mind. I'm an idiotic :-)

Reply:
The `.:` syntax is equivalent to calling the .ctor, not the Between method.